src/SupervisorOptions.php

```diff
@@ -176,7 +184,8 @@ public function __construct($name,
                                 $force = false,
                                 $nice = 0,
                                 $balanceCooldown = 3,
-                                $balanceMaxShift = 1)
+                                $balanceMaxShift = 1,
+                                $parentId = 0)
```

Comment:
@halaei why didn't you implement this dependency? Atm it'll always be zero.

Reply:
Sorry @driesvints It seems I made a mistake. Setting parentId in constructor also doesn't work because it is still zero in ensureParentIsRunning.

Reply:
@halaei what do you mean? If we fix this by passing the `$parentId` from the constructor to the property it should work, no? Why do you think it would still be zero?

Reply:
@driesvints I just dump() the value! It was 0 for sure. I sent a draft PR